nightfall/detection_rules.py

```diff
@@ -0,0 +1,152 @@
+from nightfall.exceptions import NightfallUserError
+
+
+class DetectionRule:
+    def __init__(self, detectors: list, logical_op: str = "ANY"):
```

Comment:
does python have a notion of enums? should we use them for logical ops? (and also confidence)